stock_location_archive_constraint/models/product_template.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.constrains("active")
+    def _check_active(self):
```

Comment:
Why not call to _check_active method of template variants...
variants = self.product_variant_ids._check_active()
??